### ISSUE TITLE
C++ API 'nn::Sequential' has inconsistant behavior with python version

### ISSUE THREAD

**Issue by FluorineDog** [pytorch/pytorch#19499]
## To Reproduce
C++ version
```c++
class FakeNet : public torch::nn::Module {
  public:
    FakeNet() {
        id = register_module("id", nn::Sequential());
    }
    torch::Tensor forward(torch::Tensor x) {
        // Runtime Error: Cannot call forward() on an empty Sequential
        return id->forward(x); 
    }
  private:
    nn::Sequential id;
}; 
```
Python Version
```python
class FakeNet(nn.Module):
    def __init___(self):
        self.id = nn.Sequential() 
    def forward(self, x):
        x = id(x) # OK
        return x
```
`nn::Sequential` doesn't overload `operator()` like other functions (e.g., `nn::Conv2d, nn::BatchNorm`), while in its `forward` function, it has an `AT_CHECK` on `is_empty() == false`, which makes using `nn::Sequential` as identity function impossible.  

## Suggestion
The C++ version should implement `nn::Sequential::operator()(...)`, which should silently return x when `is_empty() == true`.
  
## Environment
 - PyTorch Version (e.g., 1.0.1):

**Comment by ailzhang** [comment]
Yea our cpp API is not consistent with python api on this. Note that your python example was actually calling python `id()` instead of `self.id()`. 
A proper python example is: 
```
import torch
import pdb
import torch.nn as nn


class FakeNet(nn.Module):
    def __init__(self):
        super(FakeNet, self).__init__()
        # self.conv = nn.Conv2d(1, 1, 1)
        self.conv = nn.Sequential()

    def forward(self, x):
        y = self.conv(x)
        return y


input = torch.rand(1, 1, 2, 2)
pdb.set_trace()
m = FakeNet()
print(m(input))
```
We should match the behavior I think.

**Comment by ailzhang** [comment]
We're happy to accept an PR for it, would you minding sending one? @FluorineDog

**Comment by FluorineDog** [comment]
Gladly, I will have a look tomorrow, but I have to go to bed now :)
BTW, I'm working on a C++ demo on ResNet50, currently runnable, but it still needs to be refined. I will send a PR when I'm done. 
Cheers.
